lib/output.js

```diff
@@ -887,6 +904,21 @@ function OutputStream(options) {
         self._do_print(output, "throw");
     });
 
+    /* -----[ yield ]----- */
+
+    DEFPRINT(AST_Yield, function(self, output){
+        var star = self.is_star ? "*" : "";
+        output.print("yield" + star);
+        if (self.expression) {
+            output.space();
+            self.expression.print(output);
+        } else if (self.is_star === true) {
```

Comment:
I'll take the risk of always having an expression if is_star is set. I've added some documentation on AST_Yield as well as insurance, but that will only work with "well behaved" developers.